scripts/pages/zoning/zoning.ts

```diff
@@ -254,48 +257,52 @@ class ZoneMenuHandler {
 		const checkpointTag = $.Localize('#Zoning_Checkpoint');
 		const cancelTag = $.Localize('#Zoning_CancelZone');
 		const endTag = $.Localize('#Zoning_EndZone');
-		for (const [i, segment] of track.zones.segments.entries()) {
-			const majorId = segment.name || `${segmentTag} ${i + 1}`;
-			const segmentChildContainer = this.addTracklistEntry(
-				trackSegmentContainer,
-				majorId,
-				TracklistSnippet.SEGMENT,
-				{
-					track: track,
-					segment: segment,
-					zone: null
-				}
-			);
+		if (track.zones?.segments?.length > 0) {
```

Comment:
Shouldn't need this check. If you're worried about the array being `null`, you can do `for (const [i, segment] of track.zones.segments?.entries() ?? [])`. Looks a bit weird, but saves a whole level of indentation.